### PR TITLE
Function otp_connect

### PR DESCRIPTION
**Pull request by marcusyoung**
Would appreciate comments on this `otp_connect()` function.

The function returns an S3 object of class `otpconnect`. This object would then be used as the first argument in all the other functions that query an OTP instance.

There are two methods for the class. One generates a URL, and the second checks whether the router is reachable. By default the connection object will only be returned if the router is reachable. This behaviour is controlled by the `check` parameter.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -13,4 +13,8 @@ Description: An interface for Open Trip Planner <http://www.opentripplanner.org/
 License: GPL-3
 Encoding: UTF-8
 LazyData: true
+Imports:
+    checkmate,
```
Comment (Robinlovelace) [inline DESCRIPTION:17]: Great to see use of a proper data checking framework. Note `stopifnot()` can also go far.

File: R/otp-connect.R [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+otp_connect <- function(hostname = "localhost",
+                   router = "default",
+                   port = 8080,
+                   ssl = FALSE,
+                   check = TRUE)
+{
+  # argument checks
+
+  coll <- checkmate::makeAssertCollection()
+  checkmate::assert_string(hostname, add = coll)
```
Comment (Robinlovelace) [inline R/otp-connect.R:34]: OK great, sensible use of checkmate.

**Review by Robinlovelace — APPROVED** [review]
Looks good to me.

**Comment by marcusyoung** [comment]
Great, thanks for the feedback.

**Pull request merged**